presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralInterpreter.java

```diff
@@ -169,15 +170,18 @@ else if (value.equals(Float.POSITIVE_INFINITY)) {
             }
         }
 
-        if (type instanceof VarcharType) {
+        if (type instanceof VarcharType || type instanceof CharType) {
```

Comment:
I see that code is pretty similar, but I think it would look much easier if you would add another `if` block.

Reply:
Done